### PR TITLE
Added another advantage

### PR DESCRIPTION
**Pull request by eskimor**
- which if find compelling. That is actually one of the most interesting aspects of the concept to me.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ Please read the [paper](http://okmij.org/ftp/Haskell/extensible/exteff.pdf) for
 
   * Effects can be added, removed, and interwoven without changes to code not
     dealing with those effects.
+  * You get a better decoupling between an implementation and its interface.
+    For example whether or not you rely on IO for implementing a certain effect or not,
+    is completely transparent to the user.
```
Comment (suhailshergill) [inline README.md]: @eskimor i'm not sure i follow. isn't this always transparent given haskell's type system whether or not we use `extensible-effects`?

**Comment by eskimor** [comment]
Not really. If you have a typeclass for some functionality and and implementation might need IO then all implementations automatically live in IO, because the class' method would have to live in IO. Consider for example the PersistStore backend type class here:
http://hackage.haskell.org/package/persistent-2.2.4/docs/Database-Persist-Class.html
all functions have the constraint MonadIO, since an implementation might need IO. In contrast, my wrapper using the freer package:

https://github.com/gonimo/gonimo-back/blob/master/src/Gonimo/Database/Effects.hs

the functions (obviously) don't depend on IO, regardless whether the implementation does or not. The used interpreter might use IO or not - it does not matter.

This is just an example which illustrates the strong decoupling of interface to implementation you get with extensible-effects.

Another nice boon of extensible-effects to mtl is that you get interfaces without the need for type classes, therefore you are completely avoiding the need for orphan instances - you can define an interpreter everywhere! I think extensible-effects is just really, really awesome and I thought I highlight some of the advantages which struck me during its usage.

**Comment by eskimor** [comment]
But in any case if I phrased it so poorly that even the author of the library does not understand what I mean, I'll probably better close this PR ;-)

**Comment by suhailshergill** [comment]
@eskimor but you more than made up for it in the explanation above. i think those are valid points, and worth a mention. i think the current changeset requires a bit of tweaking, but we can work through it. am reopening the PR. feel free to push tweaks, otherwise i'll get to tweaking it after i'm done with #56

**Comment by eskimor** [comment]
Jeah #56 :-) I will think about it and submit a PR when I came to a better phrasing!

**Comment by eskimor** [comment]
Closing: It is not true, you can do the same with transformers. E.g.

https://lexi-lambda.github.io/blog/2016/06/12/four-months-with-haskell/

**Pull request closed without merging**